test/VoltzEndToEnd.test.ts

```diff
@@ -1004,4 +1004,180 @@ contract<{}, DeployOptions, CustomContext>("Voltz E2E", function () {
             1000
         );
     });
+
+    const doVariableTrade = async () => {
+        // trade VT with some other account
+        const { test } = await getNamedAccounts();
+        const testSigner = await hre.ethers.getSigner(test);
+        await mint(
+            "USDC",
+            testSigner.address,
+            BigNumber.from(10).pow(6).mul(1000000)
+        );
+        await this.usdc
+            .connect(testSigner)
+            .approve(this.periphery, BigNumber.from(10).pow(27));
+        await this.peripheryContract.connect(testSigner).swap({
+            marginEngine: this.marginEngine,
+            isFT: false,
+            notional: BigNumber.from(10).pow(6).mul(1000000),
+            sqrtPriceLimitX96: MIN_SQRT_RATIO.add(1),
+            tickLower: -60,
+            tickUpper: 60,
+            marginDelta: BigNumber.from(10).pow(6).mul(1000000),
+        });
+    };
+
+    const doFixedTrade = async () => {
+        // trade FT with some other account
+        const { test } = await getNamedAccounts();
+        const testSigner = await hre.ethers.getSigner(test);
+        await mint(
+            "USDC",
+            testSigner.address,
+            BigNumber.from(10).pow(6).mul(1000000)
+        );
+        await this.usdc
+            .connect(testSigner)
+            .approve(this.periphery, BigNumber.from(10).pow(27));
+        await this.peripheryContract.connect(testSigner).swap({
+            marginEngine: this.marginEngine,
+            isFT: true,
+            notional: BigNumber.from(10).pow(6).mul(1000000),
+            sqrtPriceLimitX96: MAX_SQRT_RATIO.sub(1),
+            tickLower: -60,
+            tickUpper: 60,
+            marginDelta: BigNumber.from(10).pow(6).mul(1000000),
+        });
+    };
+
+    const testFunction = async (swapsType1: number, swapsType2: number) => {
+        expect(await this.usdc.balanceOf(this.user1.address)).to.be.eq(0);
+        expect(await this.usdc.balanceOf(this.user2.address)).to.be.eq(0);
+
+        await mint(
+            "USDC",
+            this.user1.address,
+            BigNumber.from(10).pow(6).mul(100000)
+        );
+        await this.preparePush();
+        await this.erc20RootVault
+            .connect(this.user1.signer)
+            .deposit(
+                [BigNumber.from(10).pow(6).mul(100000)],
+                BigNumber.from(0).toString(),
+                []
+            );
+        this.user1.lpTokens = await this.erc20RootVault.balanceOf(
+            this.user1.address
+        );
+
+        if (swapsType1 == 1) {
+            await doVariableTrade();
+            let currentPosition = await this.voltzVault.currentPosition();
```

Comment:
Here you rebalance in the same position, hence your liquidity is not used for the second swap (implying that it doesn't generate any fees/cashflow there).

This is one way of rebalancing:
```
if (swapsType1 == 1) {
            await doVariableTrade();
            const currentTick = await this.peripheryContract.getCurrentTick(this.marginEngine);
            await this.voltzVault.connect(this.voltzVaultOwner).rebalance({
                tickLower: Math.floor((currentTick - 7000) / 60) * 60,
                tickUpper: Math.floor((currentTick + 7000) / 60) * 60,
            });
        } else if (swapsType1 == -1) {
            await doFixedTrade();
            const currentTick = await this.peripheryContract.getCurrentTick(this.marginEngine);
            await this.voltzVault.connect(this.voltzVaultOwner).rebalance({
                tickLower: Math.floor((currentTick - 7000) / 60) * 60,
                tickUpper: Math.floor((currentTick + 7000) / 60) * 60,
            });
        } else {
            // do nothing
        }
```